app/services/remark_generators/payment_checker.rb

```diff
@@ -0,0 +1,26 @@
+module RemarkGenerators
+  class PaymentChecker
+    class << self
+      def call(cash_transactions:, regular_transactions:, outgoings:)
+        priority_debt_checker(cash_transactions:, regular_transactions:, outgoings:)
+      end
+
+    private
+
+      def priority_debt_checker(cash_transactions:, regular_transactions:, outgoings:)
+        priority_debt_cash_transactions = cash_transactions.by_operation_and_category(:debit, :priority_debt_repayment)
+        priority_debt_regular_transactions = regular_transactions.with_operation_and_category(:debit, :priority_debt_repayment)
```

Comment:
these feel like methods now (rather than caller having to know the operation type)